TechnicalDocuments/Proposals/IdentifiersForUIAutomation.md

```diff
@@ -0,0 +1,174 @@
+# Improve discoverability of elements in UI automation tests
+
+* Author(s): Anders Ha
+* Review Manager: David Rodrigues, Sergey Shulga
+
+## Introduction
+At the moment, there is no systematic approach to apply accessibility identifiers in Bento. Some components accept a custom accessibility identifier as a temporary measure, e.g. Chatbot UI components.
+
+## Motivation
+`String` accessibility identifier is fragile, and there were already a few breakages observed due to accessibility identifier mismatch. This is because the main codebase does not publish accessibility identifiers used, and therefore automation tests have no mean to ensure themselves always pick up the same identifier as defined in the main codebase. In other words, automation tests are prone to breakage when changes in accessibility identifiers happen.
```

Comment:
```suggestion
Using textual content of UI components rather than accessibility identifier is fragile, and there were already a few breakages observed due to strings mismatch. This is because the main codebase does not publish accessibility identifiers used, and therefore automation tests have no mean to ensure themselves always pick up the same identifier as defined in the main codebase. In other words, automation tests are prone to breakage when changes in UI happen.
```